packages/running-extension/src/kernels.tsx

```diff
@@ -41,14 +45,76 @@ export async function addKernelRunningSessionManager(
   const { runningChanged, RunningKernel } = Private;
   const throttler = new Throttler(() => runningChanged.emit(undefined), 100);
   const trans = translator.load('jupyterlab');
+  const shutdownUnusedLabel = trans.__('Shut Down Unused');
+  let shutdownUnusedEnabled = false;
+  const shutdownUnusedThrottler = new Throttler(
+    checkShutdownUnusedEnabled,
+    10000
+  );
 
   // Throttle signal emissions from the kernel and session managers.
-  kernels.runningChanged.connect(() => void throttler.invoke());
+  kernels.runningChanged.connect(() => {
+    void throttler.invoke();
+    void shutdownUnusedThrottler.invoke();
+  });
   sessions.runningChanged.connect(() => void throttler.invoke());
 
   // Wait until the relevant services are ready.
   await Promise.all([kernels.ready, kernelspecs.ready, sessions.ready]);
 
+  function getUnusedKernels() {
+    // Identifies unused kernels
+    return Array.from(kernels.running()).filter(
+      kernel => (kernel.connections ?? 1) < 1
+    );
+  }
+
+  async function checkShutdownUnusedEnabled() {
+    const wasEnabled = shutdownUnusedEnabled;
+    shutdownUnusedEnabled = (await getUnusedKernels()).length > 0;
+    if (wasEnabled !== shutdownUnusedEnabled) {
+      commands.notifyCommandChanged(SHUTDOWN_UNUSED_BUTTON_CLASS);
+    }
+  }
+
+  commands.addCommand(SHUTDOWN_UNUSED_BUTTON_CLASS, {
+    caption: args => (args.toolbar ? '' : shutdownUnusedLabel),
```

Comment:
To address:

```
    [documentation] › test/documentation/commands.test.ts:7:5 › All commands must have a default label 
```

What was the motivation to change it to caption in the first place?

```suggestion
  commands.addCommand(SHUTDOWN_UNUSED_BUTTON_CLASS, {
    label: args => (args.toolbar ? '' : shutdownUnusedLabel),
    caption: args => (args.toolbar ? '' : shutdownUnusedLabel),
```

Reply:
I though we wanted just a icon and no text?
so caption removes the text and keeps showing `Shut Down Unused` on hover

# caption
![cap](https://github.com/user-attachments/assets/cfc579ea-2361-45fd-b46d-8fe78bd84679)

# label
![lab](https://github.com/user-attachments/assets/0eecb396-60d3-4e96-9735-90b5a4558bed)

Reply:
Yes, we want to hide the label on the button level but we always need to keep the label on command level. I will look into this and push some commits to get it over the finish line.

Reply:
When changing it to `label` locally it works as expected:

![image](https://github.com/user-attachments/assets/c4f6c666-d424-43a1-a4c5-0af4a2cf868c)

This is because `args.toolbar` checks whether it is in a toolbar and hides the label there accordingly.